portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,44 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Chooses a random project and updates the following: 
+ * img element with the project's img URl 
+ * heading element with the project's name
+ * paragraph element with the project's description
+ * link element with the project's Github link
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function addRandomProject() {
+  const projects =
```

Comment:
You could populate this in an `init()` method, so you don't have to create it every time someone clicks on your button:

```
let init = false, projects, projectNameEl;

function init() {
 projects = [...];
 projectNameEl = document.getElementById('project-name');
 init = true;
}

```

This way you can also cache things like the elements, eliminating the need to do a `getElementById()`/`querySelector()` repeatedly, which improves performance.

```
addRandomProject() {
  if (!init) {
    init();
  }

  ...
}
```